File: src/main/java/com/ocelot/vehicle/jei/plugin/workstation/VehicleIngredient.java
package com.ocelot.vehicle.jei.plugin.workstation;

import com.mrcrayfish.vehicle.block.BlockVehicleCrate;
import com.mrcrayfish.vehicle.entity.EngineTier;
import com.mrcrayfish.vehicle.entity.EntityVehicle;
import com.mrcrayfish.vehicle.entity.WheelType;
import net.minecraft.item.ItemStack;
import net.minecraftforge.fml.common.registry.EntityEntry;
import org.apache.commons.lang3.Validate;

import java.util.Objects;

/**
 * @author Ocelot
 */
public class VehicleIngredient
{
    public final EntityVehicle vehicle;
    private final EntityEntry vehicleEntry;
    public final ItemStack crate;

    VehicleIngredient(EntityVehicle vehicle, EntityEntry vehicleEntry)
    {
        Validate.notNull(vehicleEntry.getRegistryName(), "Registry name of Vehicle {} cannot be null!", vehicle.getClass());
        this.vehicle = vehicle;
        this.vehicleEntry = vehicleEntry;
        this.crate = BlockVehicleCrate.create(vehicleEntry.getRegistryName(), 0, EngineTier.WOOD, WheelType.STANDARD, 0);
    }

    @Override
    public boolean equals(Object other)
    {
        if (this == other) return true;
        if (!(other instanceof VehicleIngredient)) return false;
        return this.vehicleEntry.equals(((VehicleIngredient) other).vehicleEntry);
    }

    @Override
    public int hashCode()
    {
        return Objects.hash(this.vehicleEntry);
    }

    @Override
    public String toString()
    {
        return "VehicleIngredient{vehicleEntry=" + this.vehicleEntry.getRegistryName() + "}";
    }
}

File: src/main/java/com/ocelot/vehicle/jei/plugin/workstation/VehicleIngredient.kt
package com.ocelot.vehicle.jei.plugin.workstation

import com.mrcrayfish.vehicle.block.BlockVehicleCrate
import com.mrcrayfish.vehicle.entity.EngineTier
import com.mrcrayfish.vehicle.entity.EntityVehicle
import com.mrcrayfish.vehicle.entity.WheelType
import net.minecraftforge.fml.common.registry.EntityEntry

class VehicleIngredient(val vehicle: EntityVehicle, val vehicleEntry: EntityEntry) {

    private val registryName = vehicleEntry.registryName!!
    val crate = BlockVehicleCrate.create(registryName, 0, EngineTier.WOOD, WheelType.STANDARD, 0)!!

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as VehicleIngredient

        return vehicleEntry == other.vehicleEntry
    }

    override fun hashCode(): Int {
        return vehicleEntry.hashCode()
    }

    override fun toString(): String {
        return "VehicleIngredient(vehicle=$registryName)"
    }
}
